### PR TITLE
fix: Manual page slug was overwritten when changing page title

### LINKED ISSUE #8795: [BUG] Changing title on page creation discards slug
## Description

A manually written slug is discarded when changing the title.

## Steps to reproduce

1. Log in
2. Create new page
3. Set page title to `I AM A TILE`
4. Set slug to `some-page`
5. Set page title to `I AM A TITLE`
6. Slug is changed to `i-am-a-title`

## Expected behaviour

I would not expect the title change to be reflected into the slug if I provided a custom one. I also believe this has been different in previous versions if I remember correctly.

## Actual behaviour

My custom slug is discarded.

## Screenshots

None attached, the reproducible steps should hopefully be sufficient.

## Additional information (CMS/Python/Django versions)

Django CMS 3.11.11

## Do you want to help fix this issue?

* [ ] Yes, I want to help fix this issue and I will join the channel #pr-reviews on [the Discord Server](https://discord-pr-review-channel.django-cms.org) to confirm with the community that a PR is welcome.
* [x] No, I only want to report the issue.

### PR DESCRIPTION
**Pull request by fsbraun**
## Summary by Sourcery

Preserve user-controlled page slugs while maintaining reliable automatic slug generation and frontend test execution.

Bug Fixes:
- Preserve manually entered or existing page slugs when page titles change, while allowing automatic generation to resume when the slug is emptied.
- Avoid conflicts with Django admin slug prepopulation and prevent duplicate slug handlers on the same field.
- Resolve Chrome executable paths across supported Puppeteer versions in frontend CI and test configuration.

Enhancements:
- Improve slug field and title-field discovery for prefixed and nested forms.

Build:
- Update frontend development and testing dependencies.

Tests:
- Add frontend unit tests covering automatic slug generation, manual overrides, admin prepopulation, duplicate binding, change tracking, and Unicode handling.

## Related resources

* fixes #8795 
* #...

## Checklist

* [ ] I have opened this pull request against ``main``
* [ ] I have added or modified the tests when changing logic
* [ ] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [ ] I have read the [contribution guidelines](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined [our Discord Server](https://discord-pr-review-channel.django-cms.org) and the channel [#pr-reviews](https://discord.com/channels/800813886689247262/1236299181761630249) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: cms/static/cms/js/modules/slug.js [hunk 1]
```diff
@@ -1,47 +1,74 @@
...
     }
+    let fields;
 
-    let prefill = false;
+    try {
+        fields = JSON.parse(constants.dataset.prepopulatedFields || '[]');
+    } catch {
+        return false;
+    }
+    return fields.some(field => field.id && document.querySelector(field.id) === slug);
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply cms/static/cms/js/modules/slug.js:18]: @sourcery-ai this assertion is wrong

**Review by vinitkumar — COMMENTED** [review]
Looks good to me. Let's get this merged and released :)
